### PR TITLE
DDCNL-7195: Delete API for enrolment

### PR DESCRIPTION
**Pull request by Christian-QA**

### CODE REVIEW

File: it/helpers/TestITData.scala [hunk 1]
```diff
@@ -29,6 +29,7 @@ import uk.gov.hmrc.taxenrolmentassignmentfrontend.models._
 object TestITData {
 
   val NINO: String = "JT872173A"
+  val mismatchNino: String = "testFakeNino" // TODO - Add NINO generator
```
Comment (pascal22p) [inline it/helpers/TestITData.scala:32]: comment?

File: it/helpers/TestITData.scala [hunk 2]
```diff
@@ -29,6 +29,7 @@ import uk.gov.hmrc.taxenrolmentassignmentfrontend.models._
 object TestITData {
 
   val NINO: String = "JT872173A"
```
Comment (pascal22p) [inline it/helpers/TestITData.scala:31]: Use a random nino

File: it/helpers/TestITData.scala (hunk 1 above)
Comment (pascal22p) [reply it/helpers/TestITData.scala:32]: No need for a generator but a random nino

File: test/uk/gov/hmrc/taxenrolmentassignmentfrontend/helpers/TestData.scala [hunk 3]
```diff
@@ -276,6 +276,18 @@ object TestData {
       .withSession("sessionId" -> "FAKE_SESSION_ID")
       .withFormUrlEncodedBody(data.toSeq: _*)
 
+  val userDetailsWithMismatchNino: UserDetailsFromSession =
+    UserDetailsFromSession(
+      CREDENTIAL_ID,
+      "testFakeNino", // TODO - Add NINO generator
```
Comment (pascal22p) [inline test/uk/gov/hmrc/taxenrolmentassignmentfrontend/helpers/TestData.scala:282]: comment?

**Review by pascal22p — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/actions/PTMismatchCheckAction.scala [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+                        .url
+                    )
+                  )
+                case None =>
+                  val ex = new RuntimeException(s"Redirect url is missing from the query string")
+                  logger.error(ex.getMessage, ex)
+                  block(request)
+              }
+            case _ =>
+              block(request)
```
[the remark this answers is not in the document]
Comment (Christian-QA) [reply app/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/actions/PTMismatchCheckAction.scala:71]: Done

File: it/helpers/TestITData.scala (hunk 1 above)
Comment (Christian-QA) [reply it/helpers/TestITData.scala]: Done

File: test/uk/gov/hmrc/taxenrolmentassignmentfrontend/helpers/TestData.scala (hunk 3 above)
Comment (Christian-QA) [reply test/uk/gov/hmrc/taxenrolmentassignmentfrontend/helpers/TestData.scala]: Done

**Review by pascal22p — CHANGES_REQUESTED** [review]
Insufficient coverage in the integration test. A case is missing when the deletion of the enrolment failed.

**Review by pascal22p — APPROVED** [review]

**Pull request merged**